mkosi/run.py

```diff
@@ -407,7 +407,7 @@ def apivfs_cmd(root: Path) -> list[PathString]:
 def chroot_cmd(root: Path, *, options: Sequence[PathString] = ()) -> list[PathString]:
     cmdline: list[PathString] = [
         # No exec here because we need to clean up the /work directory afterwards.
-        "sh", "-c", f"mkdir --mode=777 {root / 'work'} && $0 \"$@\" && rm -rf {root / 'work'}",
+        "sh", "-c", f"trap \"rm -rf {root / 'work'}\" EXIT && mkdir -p --mode=777 {root / 'work'} && $0 \"$@\"",
```

Comment:
I think we can avoid the double quoting here
```suggestion
        "sh", "-c", f"trap 'rm -rf {root / 'work'}' EXIT && mkdir -p --mode=777 {root / 'work'} && $0 \"$@\"",
```